api/apps/api/src/modules/scenario-specification/__tests__/determine-features/scenario-specification-determine-features.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { getFixtures } from './fixtures';
+
+let fixtures: ReturnType<typeof getFixtures>;
+
+beforeEach(() => {
+  fixtures = getFixtures();
+});
+
+describe(`Draft specification with non-calculated features determined`, () => {
+  beforeEach(() => {
+    fixtures.GivenDraftWasSubmitted();
+    fixtures.WhenFeaturesWereDetermined(false);
+  });
+
+  test(`Then specification isn't published`, () => {
+    fixtures.ThenSpecificationIsNotPublished();
+  });
+});
+
+describe(`Draft specification with all-calculated features determined`, () => {
+  beforeEach(() => {
+    fixtures.GivenDraftWasSubmitted();
+    fixtures.WhenFeaturesWereDetermined(true);
+  });
+
+  test(`Then specification isn't published`, () => {
+    fixtures.ThenSpecificationIsNotPublished();
+  });
+});
+
+describe(`"Created" specification with non-calculated features determined`, () => {
+  beforeEach(() => {
+    fixtures.GivenCreatedWasSubmitted();
+    fixtures.WhenFeaturesWereDetermined(false);
+  });
+
+  test(`Then specification is published`, () => {
+    fixtures.ThenSpecificationIsPublished();
+  });
+});
+
+describe(`"Created" specification with features determined (proved to be already calculated)`, () => {
+  beforeEach(() => {
+    fixtures.GivenCreatedWasSubmitted();
+    fixtures.WhenFeaturesWereDetermined(true);
+  });
+
+  test(`Then specification is activated`, () => {
+    fixtures.ThenSpecificationIsActivated();
+    fixtures.ThenSpecCandidateIsEmpty();
```

Comment:
does it have any observable behavior?

Reply:
e.g. can't be activated again or something like that – with checking it that way you don't need to rely on snapshots